File: ruby/primes.rb
def isPrime(number)
  for i in 2..number - 1
		if number % i == 0 then
			return false
		end
	end
        return true
end

counter = 0
for i in 2..100000

  if isPrime(i) then
		counter = counter + 1
	end
end

puts "primes : " + counter.to_s

File: perl/primes.pl
#!/usr/bin/perl

use 5.010;
use strict;
use warnings;

sub isPrime
{
  my $number = shift;
	my $l = 2;
	for ($l = 2; $l < $number; $l++)
	{
		if ($number % $l == 0)
			{
				return 0;
			}
	}
	return 1;
}


my $k = 2;
my $counter = 0;
for ($k = 2; $k < 100000; $k++)
{
	if (isPrime($k))
	{
		  $counter++;
	}
}
print $counter;
